test/pytest/test_browser.py

```diff
@@ -41,11 +43,10 @@ async def spawn_test_server() -> AsyncIterator[URL]:  # noqa:RUF029
     address = os.read(addr_r, 1000).decode()
     os.close(addr_r)
 
-    try:
-        yield URL(address)
-    finally:
-        server.kill()
-        server.wait()
+    yield URL(address)
+
+    server.kill()
+    server.wait()
 
```

Comment:
This reverts the previous PR, which looks like a git accident.

Reply:
Definitely.